### PR TITLE
Remove ci jobs default

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb Ready for review after #499

File: Rakefile [hunk 1]
```diff
@@ -6,18 +6,23 @@ Rake.application.load_imports
 
 task('metrics:mutant').clear
 namespace :metrics do
-  task :mutant => :coverage do
-    success = Kernel.system(*%w[
+  task mutant: :coverage do
+    jobs = 4 if ENV.key?('CIRCLE_CI')
```
Comment (dkubb) [inline Rakefile]: This appears redundant given the code below.

Comment (mbj) [inline Rakefile]: Yeah, refactoring leftover. Good catch.

File: Rakefile [hunk 2]
```diff
@@ -6,18 +6,23 @@ Rake.application.load_imports
...
       --since HEAD~1
-      --ignore-subject Mutant::Meta*
-      --
-      Mutant*
-    ]) or fail 'Mutant task is not successful'
+      --require mutant
+      --use rspec
+      --zombie
+    ]
+    arguments.concat(%W[--jobs 4]) if ENV.key?('CIRCLE_CI')
```
Comment (dkubb) [inline Rakefile]: Have you considered having a default of `0` jobs indicate the number of cores should be chosen? This is the vonention used by parallel. If you adopted those conventions, then maybe you could have `--jobs #{ENV.key?('CIRCLE_CI') ? 4 : 0}`

Comment (mbj) [inline Rakefile]: I do not like this inline signalling. Assinging `0` a special behavior seems to be weird. If you right now boot mutant with `--jobs 0` the system behaves like you'd expect: Boots but does not make progress, as no workers where spawned.

Comment (mbj) [inline Rakefile]: I simply do not like "C-Style" APIs where integers heavily get abused for control flow triggers. Maybe mutant could support `--jobs detect` or something instead that makes it super clear.

**Pull request merged**